tee-worker/omni-executor/rpc-server/src/methods/pumpx/notify_limit_order_result.rs

```diff
@@ -0,0 +1,88 @@
+use crate::{
+	error_code::*, oneshot, server::RpcContext, verify_auth::verify_auth, Decode, Deserialize,
+	ErrorCode,
+};
+use executor_core::native_task::*;
+use executor_primitives::OmniAuth;
+use heima_primitives::{Identity, Web2IdentityType};
+use jsonrpsee::{types::ErrorObject, RpcModule};
+use native_task_handler::{NativeTaskError, NativeTaskOk, NativeTaskResponse};
+
+#[derive(Debug, Deserialize)]
+pub struct NotifyLimitOrderResultParams {
+	pub user_email: String,
+	pub intent_id: u32,
+	pub result: String,
+	pub message: Option<String>,
+	pub auth_token: String,
+}
+
+impl From<NotifyLimitOrderResultParams> for NativeTaskWrapper<NativeTask> {
+	fn from(p: NotifyLimitOrderResultParams) -> Self {
+		Self {
+			task: NativeTask::PumpxNotifyLimitOrderResult(
+				Identity::from_web2_account(p.user_email.as_str(), Web2IdentityType::Email),
+				p.intent_id,
+				p.result,
+				p.message,
+			),
+			nonce: None,
+			auth: Some(OmniAuth::AuthToken(p.auth_token)),
+		}
+	}
+}
+
+pub fn register_notify_limit_order_result(module: &mut RpcModule<RpcContext>) {
+	module
+		.register_async_method("pumpx_notifyLimitOrderResult", |params, ctx, _| async move {
+			let internal_error: ErrorObject = ErrorCode::InternalError.into();
+			let params = params.parse::<NotifyLimitOrderResultParams>()?;
+
+			let wrapper: NativeTaskWrapper<NativeTask> = params.into();
+			if wrapper.task.require_auth() && verify_auth(ctx.clone(), &wrapper).await.is_err() {
+				return Err(ErrorCode::ServerError(AUTH_VERIFICATION_FAILED_CODE).into());
+			}
```

Comment:
We have to additionally check the token is the `access` type token - see: https://github.com/litentry/heima/blob/5324e977eb4f5c3df63285b9cfbc2543ffc8f0e9/tee-worker/omni-executor/rpc-server/src/methods/pumpx/sign_limit_order.rs#L76-L85

This is not perfect coding style - we'll improve it later